File: Serene/Serene.Script/BasicSamples/CancellableBulkActionGrid.cs

namespace Serene.BasicSamples
{
    using jQueryApi;
    using Serenity;
    using System.Collections.Generic;

    public class CancellableBulkActionGrid : Northwind.OrderGrid
    {
        private GridRowSelectionMixin rowSelection;

        public CancellableBulkActionGrid(jQueryObject container)
            : base(container)
        {
        }

        protected override void CreateToolbarExtensions()
        {
            base.CreateToolbarExtensions();

            rowSelection = new GridRowSelectionMixin(this);
        }

        protected override List<ToolButton> GetButtons()
        {
            return new List<ToolButton>
            { 
                new ToolButton
                {
                    Title = "Perform Bulk Action on Selected Orders",
                    CssClass = "send-button",
                    OnClick = delegate
                    {
                        if (!this.OnViewSubmit())
                            return;

                        var action = new OrderBulkAction();
                        action.Done += rowSelection.ResetCheckedAndRefresh;
                        action.Execute(rowSelection.GetSelectedKeys());
                    }
                }
            };
        }

        protected override List<SlickColumn> GetColumns()
        {
            var columns = base.GetColumns();
            columns.Insert(0, GridRowSelectionMixin.CreateSelectColumn(() => rowSelection));
            return columns;
        }

        protected override SlickRemoteViewOptions GetViewOptions()
        {
            var opt = base.GetViewOptions();
            opt.RowsPerPage = 2500;
            return opt;
        }
    }
}

File: Serene/Serene.Core/Modules/BasicSamples/Grids/CancellableBulkAction/CancellableBulkActionGrid.ts
/// <reference path="../../../Northwind/Order/OrderGrid.ts" />
/// <reference path="OrderBulkAction.ts" />

namespace Serene.BasicSamples {

    @Serenity.Decorators.registerClass()
    export class CancellableBulkActionGrid extends Northwind.OrderGrid {
        private rowSelection: Serenity.GridRowSelectionMixin;

        constructor(container: JQuery) {
            super(container);
        }

        protected createToolbarExtensions() {
            super.createToolbarExtensions();
            this.rowSelection = new Serenity.GridRowSelectionMixin(this);
        }

        protected getButtons() {
            return [{
                title: 'Perform Bulk Action on Selected Orders',
                cssClass: 'send-button',
                onClick: () => {
                    if (!this.onViewSubmit()) {
                        return;
                    }

                    var action = new OrderBulkAction();
                    action.done = () => this.rowSelection.resetCheckedAndRefresh();
                    action.execute(this.rowSelection.getSelectedKeys());
                }
            }];
        }

        protected getColumns() {
            var columns = super.getColumns();
            columns.splice(0, 0, Serenity.GridRowSelectionMixin.createSelectColumn(() => this.rowSelection));
            return columns;
        }

        protected getViewOptions() {
            var opt = super.getViewOptions();
            opt.rowsPerPage = 2500;
            return opt;
        }
    }
}